client/core/core.go

```diff
@@ -494,6 +497,86 @@ func (dc *dexConnection) ack(msgID uint64, matchID order.MatchID, signable msgjs
 	return nil
 }
 
+// register submits a new 'register' request to the server. Depending on the
+// server version, either encKey or encKeyLegacy will be set, but not both.
+func (c *Core) register(dc *dexConnection, crypter encrypt.Crypter) (regRes *msgjson.RegisterResult,
+	acctPubB, encKey, encKeyLegacy []byte, paid bool, err error) {
+
+	creds := c.creds()
+
+	// register is called before the *dexConnection is in the Core.conns map,
+	// so no risk of concurrent access.
+	regV2 := dc.acct.dexPubKey != nil
+
+	var accountSuspended bool
+	for {
+		if regV2 && !accountSuspended {
+			encKey, acctPubB, err = dc.acct.setupCryptoV2(creds, crypter)
+			if err != nil {
+				return nil, nil, nil, nil, false, newError(acctKeyErr, "setupCryptoV2 error: %v", err)
+			}
+		} else {
+			encKey = nil
+			// This is an old-style DEX that doesn't provide their pubkey in the
+			// config response. Gotta use a legacy-style key.
+			encKeyLegacy, acctPubB, err = dc.acct.setupCryptoLegacy(crypter)
+			if err != nil {
+				return nil, nil, nil, nil, false, codedError(acctKeyErr, err)
+			}
+		}
+		dexReg := &msgjson.Register{
+			PubKey: acctPubB,
+			Time:   encode.UnixMilliU(time.Now()),
+		}
+		regRes = new(msgjson.RegisterResult)
+		err = dc.signAndRequest(dexReg, msgjson.RegisterRoute, regRes, DefaultResponseTimeout)
+		if err == nil {
+			break
+		}
+		var msgErr *msgjson.Error
+		if errors.As(err, &msgErr) {
+			switch msgErr.Code {
+			case msgjson.AccountExistsError:
+				// Server provides fee coin as the error message.
+				dc.acct.feeCoin, err = hex.DecodeString(msgErr.Message)
+				if err != nil {
+					err = fmt.Errorf("error decoding coin ID from message %q", msgErr.Message)
+				} else {
+					paid = true
+				}
+				return
+			case msgjson.AccountSuspendedError:
+				// Retry with a legacy-style key.
+				c.log.Infof("HD account key for %s was reported as suspended. Generating a legacy-style key instead.", dc.acct.host)
+				accountSuspended = true
+				dc.acct.resetKeys()
+				continue
+			}
```

Comment:
I'm leaning strongly toward not supporting legacy servers.  I think it's undesirable to continue creating local client accounts that are not recoverable with the seed.

Regarding the suspended path, I'm a little uncertain.  If the user can't resolve the issue, there's probably good reason for the client to have to re-initialize to get back on that server.  But even if that doesn't feel right, the derivation path can be extended for that server, and it can be iterated through until a "not suspended" or new account is found. This would be preferable to defaulting to legacy keys.

With the bond work, I'm dropping the old register protocol support entirely anyway because it has to go away server side for a few reasons.

Reply:
Oh, I also was going to suggest we push 0.2.2 with DEXPubKey in the config response